app/code/core/Mage/Sales/Model/Order/Status/History.php

```diff
@@ -69,18 +68,27 @@ protected function _construct()
     }
 
     /**
-     * Set order object and grab some metadata from it
+     * Set order object
      *
      * @param   Mage_Sales_Model_Order $order
      * @return  $this
      */
     public function setOrder(Mage_Sales_Model_Order $order)
     {
         $this->_order = $order;
-        $this->setStoreId($order->getStoreId());
         return $this;
     }
 
+    /**
+     * Get store id
+     *
+     * @throws Mage_Core_Model_Store_Exception
+     */
+    public function getStoreId(): int
+    {
+        return $this->getStore()->getStoreId();
```

Comment:
is it $this->getStore()->getStoreId() or $this->getOrder()->getStoreId()

Reply:
`getStore()`, since it checks if the Order is set first, otherwise defaults to current store.

Reply:
true! stupid question :-)